### PR TITLE
Transfers: Refactor list transfer requests

### PR DESCRIPTION
**Pull request by bziemons**
Since I was a bit annoyed by the returned tuple of `__list_transfer_requests_and_source_replicas` and floating variables to look up certain columns, I introduce a named tuple here. This should make further changes in these parts less painful.

Based on #4458

### CODE REVIEW

**Comment by rcarpa** [comment]
Discussed offline with @bziemons . I am also working on a functional refactoring in get_transfer_requests_and_source_replica. Which also depend on #4458. Once the dependency merged, we'll see with Ben to synchronize the two changes.

File: lib/rucio/core/replica.py [hunk 1]
```diff
@@ -35,8 +35,8 @@
 # - Eli Chadwick <eli.chadwick@stfc.ac.uk>, 2020
 # - Patrick Austin <patrick.austin@stfc.ac.uk>, 2020
 # - Eric Vaandering <ewv@fnal.gov>, 2020-2021
-# - Benedikt Ziemons <benedikt.ziemons@cern.ch>, 2020
-# - Radu Carpa <radu.carpa@cern.ch>, 2021
+# - Benedikt Ziemons <benedikt.ziemons@cern.ch>, 2020-2021
+# - rcarpa <2905943+rcarpa@users.noreply.github.com>, 2021
```
Comment (mlassnig) [inline lib/rucio/core/replica.py]: ?

File: lib/rucio/tests/test_preparer.py [hunk 2]
```diff
@@ -304,26 +303,24 @@ def setup(rse):
...
-        request_tuples[i][request_id_col] = i
-        request_tuples[i][distance_col] = 3 - i
-
-    result = sort_requests_minimum_distance(request_tuples)
-    assert next(result)[request_id_col] == 2
-    assert next(result)[request_id_col] == 1
-    assert next(result)[request_id_col] == 0
+    request_dicts = [{}, {}, {}]
+    for i in range(len(request_dicts)):
+        request_dicts[i]['request_id'] = i
```
Comment (mlassnig) [inline lib/rucio/tests/test_preparer.py:308]: i neither understand why this was multiplied with 21, nor why it was removed? :-D

File: lib/rucio/core/replica.py (hunk 1 above)
Comment (bziemons) [reply lib/rucio/core/replica.py]: git author… I will just revert this file, I did no real change here.

File: lib/rucio/tests/test_preparer.py (hunk 2 above)
Comment (bziemons) [reply lib/rucio/tests/test_preparer.py:308]: The multiplication to build that long tuple was just to return the same format as the refactored `__list_transfer_requests_and_source_replicas` function

**Comment by bari12** [comment]
@rcarpa please coordinate in what order to merge these PRs. and close/mark them draft accordingly :-)

**Comment by rcarpa** [comment]
I think mine should me merged first, because it is much more complex and error-prone to rebase. It changes functional behavior, not only cosmetic changes.  After that we'll rework this merge request on top of mine. As it bring a lot of additional benefits. @bziemons what is your opinion ?

**Comment by bziemons** [comment]
I agree, this is a single change and is easier to rebase.

**Comment by rcarpa** [comment]
This one should probably be clossed now ?  It was integrated into #4542

**Pull request closed without merging**